File: Data Structures & Algorithms/search-2d-matrix/submission-1.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int m = matrix.length; //Rows
        int n = matrix[0].length;//Cols
        int low = 0;
        int high = m*n-1;

        while(low<=high){
            int mid = low + (high-low)/2;
            int mid_row = mid/n, mid_col = mid%n;
            
            if (matrix[mid_row][mid_col]==target){
                return true;
            }
            else if(matrix[mid_row][mid_col]>target){
                high=mid-1;

            }
            else if (matrix[mid_row][mid_col]<target){
                low=mid+1;
            }
            else{
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-1.py
import re 
class Solution:
    def isPalindrome(self, s: str) -> bool:
        clean = re.sub(r'[^a-zA-Z0-9]', '', s).lower()
        low=0
        high=len(clean)-1

        while low<high:
            if clean[low]!= clean[high]:
                return False
            low+=1
            high-=1
        return True